### ISSUE TITLE
feature request: Delete buffer on file deletion/move

### ISSUE THREAD

**Issue by melMass** [stevearc/oil.nvim#499]
### Did you check existing requests?

- [X] I have searched the existing issues

### Describe the feature

I couldn't find if this is doable by simply hooking to some option of oil but it would be nice to have the option of automatically delete associated buffers on file deletion/move (ideally simply relink on move)

### Provide background

I usually realize on closing the editor about those hanging buffers and it confused me more than once

### What is the significance of this feature?

strongly desired

**Comment by stevearc** [comment]
Buffers should already be getting renamed when moved. Can you confirm that this is the case, or report a bug if not?

**Comment by melMass** [comment]
> Buffers should already be getting renamed when moved

That's what I recalled too it might be an isolated case (large monorepo with a relatively high LSP "action" timeout).

Sorry then, I'll try to find a way to properly reproduce

**Comment by melMass** [comment]
Yep closing.
TLDR mixing moves/deletion triggers lsp actions (rename imports etc) so those buffer become "modified" and thus aren't cleaned up. *That's a feature not a bug*
